src/main/java/uk/gov/hmcts/reform/sscs/util/OtherPartyDataUtil.java

```diff
@@ -43,6 +45,28 @@ public static void assignOtherPartyId(List<CcdValue<OtherParty>> otherParties) {
         }
     }
 
+    public static void processCaseState(SscsCaseData sscsCaseData) {
+        if (sscsCaseData.getIsFqpmRequired() == null
+                || hasDueDateSetAndOtherPartyWithoutHearingOption(sscsCaseData)) {
+            sscsCaseData.setState(State.NOT_LISTABLE);
+        } else {
+            sscsCaseData.setState(State.READY_TO_LIST);
+        }
+    }
+
+    private static boolean hasDueDateSetAndOtherPartyWithoutHearingOption(SscsCaseData sscsCaseData) {
+        return StringUtils.isNotBlank(sscsCaseData.getDwpDueDate())
+                && !everyOtherPartyHasAtLeastOneHearingOption(sscsCaseData);
+    }
+
+    private static boolean everyOtherPartyHasAtLeastOneHearingOption(SscsCaseData sscsCaseData) {
+        if (sscsCaseData.getOtherParties() != null) {
+            return sscsCaseData.getOtherParties().stream().noneMatch(otherParty -> otherParty.getValue().getHearingOptions() == null);
+        } else {
```

Comment:
less lines if you remove the else and leave the return false

Reply:
less readable imho

Reply:
no problem if you like it